### PR TITLE
Add tests for FastParser and SFDbCommandBuilder

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
Regarding issue 113

The PR adds tests for FastParser and SFDbCommandBuilder to increase code coverage

### CODE REVIEW

File: Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs
Comment (sfc-gh-ext-simba-lf) [inline Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs:48]: Previously, the new value would only be assigned if it was null or empty

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM. Wait for Michal approval.

File: Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs:47]: can we remove TODO?

File: Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs:63]: can we remove TODO?

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/SFDbCommandBuilderTest.cs:51]: It sounds to me like testing a constructor - default quotation. If really needed I would put it into a separate test.

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/SFDbCommandBuilderTest.cs:68]: isn't the next assert enough?

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/SFDbCommandBuilderTest.cs:55]: AreEqual should be enough, no?

File: Snowflake.Data.Tests/FastParserTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/FastParserTest.cs:154]: If we pass a null byte array into any FastParser methods an exception System.ArgumentNullException or NullReferenceException is thrown (depending which call has been made).
On the other hand I checked all existing calls to the parser and right now nothing like that should happen (?).
ie.:
`return FastParser.FastParseInt64(srcVal.Buffer, srcVal.offset, srcVal.length);`
which is done after null check for srcVal.
But what if srcVal.Buffer is null whereas srcVal is not null?

My suggestion is to add null checks in FastParser and write just one test showing that parser doesn't break
when one of those parser methods are done with null buffer.

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/SFDbCommandBuilderTest.cs:32]: Empty means without Data Adapter. Should we maybe move default quotations checks to a dedicated test for clarity of test purpose?

File: Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs]: Ok

File: Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Client/SnowflakeDbCommandBuilder.cs]: Ok

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/SFDbCommandBuilderTest.cs]: You're right, it's already covered in that test. It's removed

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/SFDbCommandBuilderTest.cs]: Removed

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/SFDbCommandBuilderTest.cs]: It should be enough

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/SFDbCommandBuilderTest.cs]: Yes, it should be enough

File: Snowflake.Data.Tests/SFDbCommandBuilderTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/SFDbCommandBuilderTest.cs:31]: I separated them into two tests: `TestDefaultCommandBuilder` and `TestCommandBuilderWithoutAdapter`

Is something like that what you meant?

File: Snowflake.Data.Tests/FastParserTest.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/FastParserTest.cs:163]: I added null checks and return 0 if it's null. Could you take a look?

I'm not sure if it should be returning something else

Comment (sfc-gh-mhofman) [reply Snowflake.Data.Tests/FastParserTest.cs:163]: @sfc-gh-igarish @sfc-gh-ext-simba-lf Maybe we should capture that in the parser and instead of NPE or ArgException we should throw SnowflakeDbException with internal error and a message of parsing a null buffer? 
I wonder that returning 0 seams not really a safe solution. We could introduce an error code OR throw exception mentioned above.

File: Snowflake.Data.Tests/FastParserTest.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data.Tests/FastParserTest.cs:156]: Thank you for this check @sfc-gh-ext-simba-lf

File: Snowflake.Data/Core/FastParser.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Core/FastParser.cs:92]: same as above

File: Snowflake.Data/Core/FastParser.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/FastParser.cs]: Added logging for the failure

File: Snowflake.Data/Core/FastParser.cs
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/FastParser.cs]: Added

File: Snowflake.Data/Core/FastParser.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Core/FastParser.cs:95]: thank you @sfc-gh-ext-simba-lf

File: Snowflake.Data/Core/FastParser.cs
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Core/FastParser.cs:98]: Thank you @sfc-gh-ext-simba-lf

**Pull request merged**
